mempool/src/pool/mod.rs

```diff
@@ -263,77 +269,78 @@ where
         // We don't have a notion of MAX_MONEY like Bitcoin Core does, so we also don't have a
         // `MoneyRange` check like the one found in Bitcoin Core's `CheckTransaction`
 
-        if tx.transaction().inputs().is_empty() {
-            return Err(TxValidationError::NoInputs);
-        }
+        // TODO: The following three checks may each better be discarded or moved to tx verifier
+        ensure!(
+            !tx.transaction().inputs().is_empty(),
+            TxValidationError::NoInputs,
```

Comment:
This case is already handled in verifier and it will claim that tx attempts to print money. Also in PR #796 I explicitly added such a rule for inputs.

Reply:
Cool will remove it from here then